### PR TITLE
Introduce cargo deny check to ci

### PR DESCRIPTION
**Pull request by Veetaha**
## Description

Tackles an item from https://github.com/amethyst/amethyst/issues/2407

The new job should check for RustSec advisories (there are currently two of them active for `amethyst` (#2471, #2472)
One of them is a long-standing advisory which `tokio` should fix at some point (https://github.com/tokio-rs/mio/issues/1319#issuecomment-683496712)...

This job is optional to pass because advisories are sudden and may break ci at any time.
Other things cargo-deny does is it checks for multiple versions of crates in the tree (there are a huge amount of them for now).
I have fixed some simple cases, but one of the biggest ones are the outdated `rendy` and `rand` crates.

Regarding licenses, you can see the list of them in the `deny.toml` file. There is an interesting crate here:
```
FTL (1): servo-freetype-sys@4.0.5
GPL-2.0 (1): servo-freetype-sys@4.0.5
```
It defines `license = "FTL  OR  GPL-2.0"`, not sure if this is what `amethyst` needs, I am not good at legal stuff...
## Removals

- Removed direct dependency on `itertools` crate (which was already outdated)
- Removed unused dependency `amethyst_derive` from `amethyst_assets`
- Removed unused dependency `log` from `amethyst_config`
- Removed unused dependency `fnv` from `amethyst_core`

## Modifications

- Updated `ron` crate from `v0.5` to `v0.6`. This is a **BREAKING CHANGE** since `ron` has unified it's `ser/de::Error`s to a single `Error` struct, but this error is also exposed via a public `ConfigError` enum. (See [their changelog](https://github.com/ron-rs/ron/blob/master/CHANGELOG.md#060---2020-05-21)). Surprisingly, the error changes are not mentioned in there (https://github.com/ron-rs/ron/pull/222#issuecomment-683485819)

- Updated `crossbeam-queue` from `v0.1.2` to `v0.2.2` ([changelog](https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-queue/CHANGELOG.md))

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Updated the content of the book if this PR would make the book outdated.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [ ] Ran `cargo clippy --workspace --features "empty"` (may require `cargo clean` before)
- [x] Ran `cargo build --features "empty"`
- [x] Ran `cargo test --workspace --features "empty"`

### CODE REVIEW

**Comment by valkum** [comment]
Sry for the late reply on the issues lately. 
Not sure about the mix of adding cargo-deny to the CI and changing dependencies in the same turn.
Would like to hear what the ones with merge rights think.
Else LGTM.

This would need a rebase for the CHANGELOG.md

**Comment by CleanCut** [comment]
IANAL, but I'm fairly experienced with open source licenses. I read [the FTL license](https://github.com/servo/libfreetype2/blob/f801061ee57dac0937cccffb0f7bf3fa120208f5/freetype2/docs/FTL.TXT) and I believe it is compatible with the MIT and Apache 2.0 licenses, and this dependency appears to be abiding by it.  I _think_ this is just fine.

Mixing CI and changing dependencies in the same PR isn't optimal, but I think it's fine this time.  Everything's in a bit of a crazy state right now anyway.

Let's get this merged after [the Legion stuff](https://github.com/amethyst/amethyst/issues/2517).

**Comment by jojolepro** [comment]
Hello! When you have time, please fix the conflicts and we'll merge this. Thank you!

File: amethyst_animation/src/systems/sampling.rs [hunk 1]
```diff
@@ -76,28 +75,34 @@ where
...
-                            if let Some(p) = self
-                                .inner
-                                .iter()
-                                .filter(|p| p.1 == *channel)
-                                .map(|p| p.2.clone())
-                                .last()
-                            {
-                                comp.apply_sample(channel, &p, &apply_data);
-                            }
+
```
Comment (jojolepro) [inline amethyst_animation/src/systems/sampling.rs:78]: I would really separate the dependency changes from the CI changes. If you could close this PR and make two separate ones that would be great!

**Comment by CleanCut** [comment]
@jojolepro Could you clarify which of the two strategies you you prefer be taken here? In your review you indicated both a) fixing the conflict and merging this PR, and b) closing this PR and making two new PRs, which sound mutually exclusive to me.  Given @Veetaha wrote this up 5 months ago, I'd lean towards option (a).

**Comment by ezpuzz** [comment]
thanks @Veetaha I've based #2590 on your work here.  closing in favor of that one.

**Pull request closed without merging**